lib/src/models/view/sellerprofile.dart

```diff
@@ -0,0 +1,56 @@
+import "package:btc_market/data.dart";
+import "package:btc_market/models.dart";
+import "package:btc_market/src/data/category.dart";
+
+/// This view model includes the logic required for the SellerProfile view
+
+class SellerProfileViewModel extends ViewModel {
+  // The profile variable will hold the object of Seller Profile
+  /// The profile will get the seller profile info from model of seller profile.
+  late final SellerProfile profile;
+
+  /// This set stores the cateogories in which the user has their products listed.
+  /// It is calculated on the basis of the product listings that the user has.
+  late final Set<Category> categories;
+
+  /// This list stores the list of products that the user currently has.
+  /// The data for this list will be fetched from the database which will have the details of the products that the user is selling.
+  /// Invoked the function to populate productListings and Categories.
+  late List<Product> productList;
+
+  @override
+  Future<void> init() async {
+    isLoading = true;
+    // get the seller's products, reviews, categories, etc
+    profile = models.user.sellerProfile!;
+    productList = await productListings;
+    categories = getCategories();
```

Comment:
```suggestion
    productList = await getProducts();
    categories = getCategories(productList);
```

See below

Reply:
I don't have the getProducts() method written. I've written the following:
`Future<List<Product>> get productListings async => <Product>[];`

Hence, I called await proudctListings;

The getCategories method doesn't need the parameter because it has the access to the instance variable productList.

Reply:
This made a lot of sense. I rechecked the code and found out that even the Products was not written properly. I have not directly updated my code to the suggestions, but made changes that follow the best practice. Please have a look at them and lmk if I need to change any further!